### DISCUSSION TITLE
Are square brackets in the description supported?

### DISCUSSION THREAD

**Discussion by ktprograms** [jotaen/xit#52] [File Format]
With the [vim-xit](https://github.com/ryanolsonx/vim-xit) plugin I'm using, if I have a square bracket in the description, the due date doesn't get highlighted.

Is this a valid part of the file spec? For example, I sometimes want multiple "levels" of brackets but I don't like using the same bracket:
```
[ ] Eat some food (Japanese [Sushi preferably]) -> 2023
```

**Comment by jotaen** [comment]
Yes, [the specification allows descriptions to contain any character](https://github.com/jotaen/xit/blob/main/Specification.md#description).

You find some examples in the [syntax guide](https://xit.jotaen.net/syntax-guide):

<img width="892" alt="Screenshot 2022-11-20 at 16 15 46" src="https://user-images.githubusercontent.com/3618384/202910201-aa324733-ccbf-4692-94bb-ee7c45a4a1a5.png">

**Reply by ktprograms** [reply]
Thanks, I missed that while glancing over the syntax guide. I've submitted pull requests (https://github.com/ryanolsonx/vim-xit/pull/14, https://github.com/sadotsoy/vim-xit/pull/4) to the two vim [x]it! plugins I know of.

Since I use Vim, I haven't tested the plugins for other code editors, so maybe you could add some extra "tokens" (due date/tags) to these checkbox examples just so it's clear that highlighting still needs to work?

**Reply by jotaen** [reply]
Well, I had actually extended the examples after you opened this thread 😉 

Good idea about the tokens, I’ve added that in:

<img width="487" alt="Screenshot 2022-11-24 at 11 12 37" src="https://user-images.githubusercontent.com/3618384/203757669-b2476b44-4944-4e56-8b57-b6e8f914b8c8.png">

**Reply by ktprograms** [reply]
> Well, I had actually extended the examples after you opened this thread 😉

Oh, so it wasn't a problem with my eyes? Cool

Out of curiosity, is the syntax guide stored on Git anywhere?

**Reply by jotaen** [reply]
> Out of curiosity, is the syntax guide stored on Git anywhere?

No, it’s not on Github. The HTML source code of the page is hand-crafted, though, so there is no automatic pre-processing/build, it’s just this single file, 1:1.

**Reply by ktprograms** [reply]
I see.

**Reply by jotaen** [reply]
~I think discussions on Github have no status (in contrast to issues), so it can’t be closed. It can just stay as it is for future reference.~

Thanks for helping to clarify this!

**Reply by ktprograms** [reply]
Oops, you saw my message before I deleted it 😆  (or maybe got a notification for it).
